contributor_docs/fes_reference_dev_notes.md

```diff
@@ -0,0 +1,456 @@
+@param  {String}        [func]    Name of the function
+@param  {Number|String} [color]   CSS color string or error type
+```
+##### Location
+core/friendly_errors/fes_core.js
+
+### `_friendlyFileLoadError()`
+##### Examples
+Example of File Loading Error:
+````JavaScript
+/// missing font file
+let myFont;
+function preload() {
+  myFont = loadFont('assets/OpenSans-Regular.ttf');
+};
+function setup() {
+  fill('#ED225D');
+  textFont(myFont);
+  textSize(36);
+  text('p5*js', 10, 50);
+};
+function draw() {};
+// FES will generate the following message in the console:
+// > 🌸 p5.js says: It looks like there was a problem loading your font. Try checking if the file path [assets/OpenSans-Regular.ttf] is correct, hosting the font online, or running a local server.[https://github.com/processing/p5.js/wiki/Local-server]
+````
+##### Description
+`_friendlyFileLoadError()` is called by the `loadX()` functions if there is an error during file loading.
+
+Generates and prints a friendly error message using key: `fes.fileLoadError.*`.
+
+This function generates and displays friendly error messages if a file fails to load correctly. It also checks and produces a warning if the size of a file is too large to load.
+
+The current version contains templates for generating error messages for `image`, `XML`, `table`, `text`, `json` and `font` files.
+
+Implemented to functions in:
+* `image/loading_displaying/loadImage()`
+* `io/files/loadFont()`
+* `io/files/loadTable()`
+* `io/files/loadJSON()`
+* `io/files/loadStrings()`
+* `io/files/loadXML()`
+* `io/files/loadBytes()`.
+
+The call sequence to `_friendlyFileLoadError` looks something like this:
+```
+_friendlyFileLoadError
+  _report
+```
+##### Syntax
+````JavaScript
+_friendlyFileLoadError(errorType, filePath)
+````
+##### Parameters
+```
+@param  {Number}  errorType   Number of file load error type
+@param  {String}  filePath    Path to file caused the error
+```
+##### Location
+core/friendly_errors/file_errors.js
+
+### `validateParameters()`
+##### Examples
+Example of a Missing Parameter:
+````JavaScript
+arc(1, 1, 10.5, 10);
+// FES will generate the following message in the console:
+// > 🌸 p5.js says: It looks like arc() received an empty variable in spot #4 (zero-based index). If not intentional, this is often a problem with scope: [https://p5js.org/examples/data-variable-scope.html]. [http://p5js.org/reference/#p5/arc]
+// > 🌸 p5.js says: It looks like arc() received an empty variable in spot #5 (zero-based index). If not intentional, this is often a problem with scope: [https://p5js.org/examples/data-variable-scope.html]. [http://p5js.org/reference/#p5/arc]
+````
+Example of a Type Mismatch:
+````JavaScript
+arc('1', 1, 10.5, 10, 0, Math.PI, 'pie');
+// FES will generate the following message in the console:
+// > 🌸 p5.js says: arc() was expecting Number for parameter #0 (zero-based index), received string instead. [http://p5js.org/reference/#p5/arc]
+````
+##### Description
```

Comment:
should description go before the examples here? I'm realizing that the pattern here matches the p5.js reference (leading with examples first), but for some of the functions (in particular fesCodeReader()) I'm not really sure what they do just from the function name.